File: C#/Min Stack/Min Stack.cs
public class Node {
    public int Value { get; set; }
    public Node Next { get; set; }
    public Node (int val) {
        Value = val;
        Next = null;
    }
}

public class MinStack {
    public Node StackTop { get; private set; }

    /** initialize your data structure here. */
    public MinStack () {

    }

    public void Push (int x) {
        Node newNode = new Node (x);
        newNode.Next = StackTop;
        StackTop = newNode;
    }

    public void Pop () {
        if (StackTop != null)
            StackTop = StackTop.Next;
    }

    public int Top () {
        return StackTop.Value;
    }

    public int GetMin () {
        if (StackTop == null) throw new Exception ("No Value");
        int minimal = StackTop.Value;
        Node ptr = StackTop.Next;
        while (ptr != null) {
            if (ptr.Value < minimal) minimal = ptr.Value;
            ptr = ptr.Next;
        }
        return minimal;
    }
}

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack obj = new MinStack();
 * obj.Push(x);
 * obj.Pop();
 * int param_3 = obj.Top();
 * int param_4 = obj.GetMin();
 */

File: Python/Min Stack/Min Stack.py
class MinStack:

    def __init__(self):
        self.minStack, self.dataStack = [], []

    def push(self, x):
        if len(self.minStack) == 0:
            self.minStack.append(x)
        else:
            self.minStack.append(min(self.minStack[-1], x))
        self.dataStack.append(x)

    def pop(self):
        self.minStack.pop()
        self.dataStack.pop()

    def top(self):
        return self.dataStack[-1]

    def getMin(self):
        return self.minStack[-1]

# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(x)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()
